net/device/IIotHub.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Iot.Thief.Device
+{
+    /// <summary>
+    /// An interface for device code to interract with hub
+    /// </summary>
+    internal interface IIotHub
```

Comment:
to be added:
- write properties
- hook to respond to registered C2D messages
- hook to respond to registered commands